server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -825,6 +837,17 @@ public IndexSettings(final IndexMetadata indexMetadata, final Settings nodeSetti
         this(indexMetadata, nodeSettings, IndexScopedSettings.DEFAULT_SCOPED_SETTINGS);
     }
 
+    /**
+     * Determines the index mode depending on settings. The 'logs' mode is only allowed if explicitly enabled by means of
+     * 'index.mode.logs.enabled'. If not explicitly enabled, 'standard' index mode is used in place of 'logs'.
+     */
+    private static IndexMode indexMode(final Settings nodeSettings, final IndexScopedSettings indexScopedSettings) {
```

Comment:
This method shouldn't be needed? The logs settings component template should either have the index mode configured or not?

Reply:
I understood that if usage of LogsDB is disabled (setting is false) we need to fallback to standard mode. Am I wrong?

Reply:
Maybe what is missing here is an error message saying "the setting is disabled so usage of index.mode 'logs' is not allowed'.

Reply:
The issue I am trying to solve here is...what happens if a user has a mapping with `index.mode = logs` and the setting is disabled? If I understand you comment the idea is to allow it no matter the setting and use the setting just to control what foes into `logs@settings`...I am a bit more restrictive.